src/main/default/classes/CoffeeMachine.cls

```diff
@@ -0,0 +1,68 @@
+public virtual class CoffeeMachine implements ICoffeeMachine {
+
+    protected Integer drinksServed;
+    protected Integer waterLevel;
+
+    public CoffeeMachine(){
+        drinksServed = 0;
+        waterLevel = 1000;
+    }
+
+    public Drink brewCoffee(Integer amount, Integer strength) {
+        if(drinksServed >= 5) {
+            throw new CoffeeMachineNotRinsedException('CoffeMachine should be rinsed first in order to serve coffee');    
+        }
+        if(amount < 20 ||
+           amount > 200) {
+            throw new OutOfRangeException('Coffe can be served only between 20ML to 200ML');
+        }
+        if(strength < 1 ||
+           strength > 5) {
+            throw new OutOfRangeException('Coffe Strenth can be served only between 1 to 5');
+        }
+        if(waterLevel < amount) {
+            throw new NoWaterException('Not enough water available in coffee machine. Please add water!');
+        }
+        waterLevel = waterLevel - amount;
+        drinksServed = drinksServed + 1;
+        return new Coffee(amount);
```

Comment:
I think your test case should fail as you do not set the `strength` of the coffee accordingly; I would expect something like: `new Coffee(amount, strength)`

(though I think the default `coffee` class might not have that property in the boilerplate code that I shared)